go.mod

```diff
@@ -1,31 +1,32 @@
 module github.com/openstack-k8s-operators/openstack-baremetal-operator
 
-go 1.24
+go 1.24.0
 
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/go-logr/logr v1.4.3
 	github.com/golang/glog v1.2.5
 	github.com/google/uuid v1.6.0
-	github.com/metal3-io/baremetal-operator/apis v0.6.3
-	github.com/onsi/ginkgo/v2 v2.20.1
-	github.com/onsi/gomega v1.34.1
+	github.com/metal3-io/baremetal-operator/apis v0.11.0
+	github.com/onsi/ginkgo/v2 v2.22.0
+	github.com/onsi/gomega v1.36.1
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.6.1-0.20250922082314-c83d83092a04
 	github.com/openstack-k8s-operators/openstack-baremetal-operator/api v0.0.0-00010101000000-000000000000
 	github.com/spf13/cobra v1.10.1
-	k8s.io/api v0.31.12
-	k8s.io/apimachinery v0.31.12
-	k8s.io/client-go v0.31.12
+	k8s.io/api v0.33.5
+	k8s.io/apimachinery v0.33.5
+	k8s.io/client-go v0.33.0
```

Comment:
@rabi @abays I think this pushed things to far. OCP 4.18 is 0.31, thats why I bumped the deps with the golang PR to this version. I think we'd have to check which github.com/metal3-io/baremetal-operator/apis is used on 4.18 and only bump to that version.

Reply:
Not sure if this is the correct way to check, but https://github.com/openshift/cluster-baremetal-operator/blob/release-4.18/go.mod#L272 looks pretty old (older than what we currently use) to me and it has k8s.io/api v0.30.1 and golang 1.22.0. However, release 4.19 has  k8s.io/api v0.31.7 which kind of maps to BMO 0.10. 0.

Reply:
OK after discussing with @stuggi, I've proposed https://github.com/openstack-k8s-operators/openstack-baremetal-operator/pull/327.